reframe/core/logging.py

```diff
@@ -165,7 +165,39 @@ def _format_time_rfc3339(timestamp, datefmt):
     return re.sub(r'(%)?\:z', tz_rfc3339, time.strftime(datefmt, timestamp))
 
 
-class RFC3339Formatter(logging.Formatter):
+class DynamicFieldFormatter(logging.Formatter):
+    def __init__(self, fmt=None, datefmt=None, style='%'):
+        super().__init__(fmt, datefmt, style)
+        self.__extras = {}
+        log_vars = re.findall(r'\%\((check_\S+?)\)s', fmt)
+        for v in set(log_vars):
+            self.__extras.setdefault(v, None)
+
+    def _fmt_check_attr(self, val):
+        if isinstance(val, list) or isinstance(val, tuple):
+            return ','.join(val)
+        else:
+            return val
+
+    def format(self, record):
+        if record.check:
+            for spec in self.__extras.keys():
+                _, *attr = spec.split('_')
+                if spec in record.__dict__.keys():
+                    continue
+
+                record.__dict__[spec] = self.__extras[spec]
+                if attr[0] == 'job' and record.check.job:
+                    self.__extras[spec] = self._fmt_check_attr(
+                        getattr(record.check.job, '_'.join(attr[1:])))
```

Comment:
We won't support `check_job_xxx` to be translated to `check.job.xxx`. For these cases, we keep the extras as we have them.